lib/sensors/HAL_Pwm_Capture.cpp

```diff
@@ -0,0 +1,88 @@
+#include "HAL_PWM.h"

```

Comment:
im curious was this generated by CubeMx? it kinda looks like it with all the GPIO initialization and all but i didnt know cubemx could generate c++ code. is there a .ioc file (cube project file) that u created?

Reply:
I created it but I used some examples from CubeMx. For example the encoder initialization was very similar to the CubeMx generatioin but the gpio initialization I had to do by myself